org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/SymbolsLabelProvider.java

```diff
@@ -117,9 +122,21 @@ public void dispose() {
 		super.dispose();
 	}
 
+	private final LoadingCache<Object /*URI | String*/, Optional<IResource>> resourceCache = CacheBuilder.newBuilder() //
+			.maximumSize(200) //
+			.expireAfterAccess(30, TimeUnit.SECONDS) //
```

Comment:
why 30 seconds?
Why not caching for the lifecycle of SymbolsLabelProvider? Also shouldn't the cache be static?

Reply:
I was also wondering about the caching strategy and why we chose 30 seconds and how that would compare to use weak keys. I had approved it without questions because I guess there might not be a best strategy for this case.